engine/src/cubos/engine/gizmos/plugin.cpp

```diff
@@ -193,9 +216,49 @@ static void drawGizmosSystem(Write<Gizmos> gizmos, Write<GizmosRenderer> gizmosR
     iterateGizmos(gizmos->screenGizmos, screenInfo, *gizmosRenderer, *deltaTime);
 }
 
+static void processInput(Write<GizmosRenderer> gizmosRenderer, Write<Gizmos> gizmos,
+                         EventReader<WindowEvent> windowEvent)
+{
+    for (const auto& event : windowEvent)
+    {
+        if (std::holds_alternative<MouseMoveEvent>(event))
+        {
+            gizmosRenderer->lastMousePosition = std::get<MouseMoveEvent>(event).position;
+        }
+        else if (std::holds_alternative<MouseButtonEvent>(event))
+        {
+            if (std::get<MouseButtonEvent>(event).button == cubos::core::io::MouseButton::Left)
+            {
+                gizmosRenderer->mousePressed = std::get<MouseButtonEvent>(event).pressed;
+            }
+        }
+        else if (std::holds_alternative<ResizeEvent>(event))
+        {
+            gizmosRenderer->resizeTexture(std::get<ResizeEvent>(event).size);
+        }
+    }
+
+    auto* texBuffer = new uint16_t[(std::size_t)gizmosRenderer->textureSize.x *
+                                   (std::size_t)gizmosRenderer->textureSize.y * (std::size_t)2U];
```

Comment:
```suggestion
                                   (std::size_t)gizmosRenderer->textureSize.y * 2U];
```

The cast shouldn't be necessary.